notification-svc/utils/encryption.ts

```diff
@@ -0,0 +1,53 @@
+import crypto, { CipherGCMTypes } from 'crypto'
+import { IAppointmentProps } from '../types/appointment.interface';
+
+const secret_iv = crypto.randomBytes(16);
+const secret_key = process.env.HASH_SECRET_KEY
+const ecnryption_method = process.env.HASH_ENCRYPTION_METHOD as CipherGCMTypes
+
+
+if (!secret_key || !secret_iv || !ecnryption_method) {
+    throw new Error('secretKey, secretIV, and ecnryptionMethod are required')
+}
+
+const key = crypto
+    .createHash('sha512')
+    .update(secret_key)
+    .digest('hex')
+    .substring(0, 32)
+
+const encryptionIV = crypto
+    .createHash('sha512')
+    .update(secret_iv)
+    .digest('hex')
+    .substring(0, 16)
+
+function encryptData(data: string) {
+    const cipher = crypto.createCipheriv(ecnryption_method, key, encryptionIV)
+    return Buffer.from(
+        cipher.update(data, 'utf8', 'hex') + cipher.final('hex')
+    ).toString('base64')
+}
+
+export default function generateAndShareHash(appointment: IAppointmentProps) {
+    const expirationDate = new Date();
+    expirationDate.setHours(expirationDate.getHours() + 2);
+    const toEncrypt = {
+        id: appointment.id,
+        expirationDate,
+    }
+    const toEncryptString = JSON.stringify(toEncrypt)
+    const hash = encryptData(toEncryptString)
+
+    fetch('http://localhost:3000/appointment/hash', {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+            hash,
+            encryptionIV,
+        })
+    })
+    return hash
```

Comment:
We don't need to call the appointment API here, but we need to pass not only the hash here but the encryptionIV as well.